src/Token/_RemoveTokenButton.tsx

```diff
@@ -0,0 +1,28 @@
+import {XIcon} from '@primer/octicons-react'
+import styled from 'styled-components'
+import {defaultTokenSize} from './TokenBase'
+import {TokenButtonProps, tokenButtonStyles, variants, getTokenButtonIconSize} from './_tokenButtonUtils'
+
+const RemoveTokenButton = styled.span.attrs<TokenButtonProps>(
+  ({borderOffset, parentTokenTag, variant, parentTokenIsInteractive, ...rest}) => {
+    delete rest.children
```

Comment:
Doing all of this inside the `styled` function feels a little off to me. What if we organized the component more like this:

```
const TokenButton = styled.button`
   ...all the styles here...
`

function RemoveTokenButton({isParentInteractive, ...}) {
  return <TokenButton as={isParentInteractive? : 'span' : 'button'} ...><XIcon /></TokenButton>
}
```